lib/signal-manager.js

```diff
@@ -0,0 +1,44 @@
+const runningProcs = new Set()
+
+const forwardedSignals = [
+  'SIGINT',
+  'SIGTERM'
+]
+
+function handleSignal (signal) {
+  for (const proc of runningProcs) {
+    proc.kill(signal)
+  }
+}
+
+function setupListeners () {
+  for (const signal of forwardedSignals) {
+    if (!process.listeners(signal).includes(handleSignal)) {
+      process.once(signal, handleSignal)
+    }
+  }
+}
+
+function cleanupListeners () {
+  if (runningProcs.size === 0) {
+    for (const signal of forwardedSignals) {
+      process.removeListener(signal, handleSignal)
+    }
+  }
+}
+
+function add (proc) {
+  runningProcs.add(proc)
+  setupListeners()
```

Comment:
Seems like we could have a module-level boolean flag to say whether we're currently in setup or not-setup mode?

Reply:
i considered it, but especially with the usage of `process.once()` for the signal handlers it's easy to get into a state where one signal is handled but another is not, and if a second child is run i wanted to be sure that we for sure had all of the necessary handlers in place.